src/lib/coaler/multialign/MultiAligner.cpp

```diff
@@ -161,12 +161,26 @@ namespace coaler::multialign {
         double currentBestAssemblyScore
             = AssemblyScorer::calculateAssemblyScore(currentBestAssembly, m_pairwiseAlignments, m_ligands);
 
+        // write queue content to vector to allow for parallel for
+        std::vector<AssemblyWithScore> assembliesList;
         while (!assemblies.empty()) {
-            LigandAlignmentAssembly currentAssembly = assemblies.top().first;
-            spdlog::info("remaining assemblies to be opimized: {}", assemblies.size());
-            double currentAssemblyScore = assemblies.top().second;
-            spdlog::info("Score before opt: {}", currentAssemblyScore);
+            assembliesList.push_back(assemblies.top());
             assemblies.pop();
+        }
+
+        // locks for shared variables
+        omp_lock_t bestAssemblyScoreLock;
+        omp_init_lock(&bestAssemblyScoreLock);
+        omp_lock_t bestAssemblyLock;
+        omp_init_lock(&bestAssemblyLock);
+
+#pragma omp parallel for shared(bestAssemblyScoreLock, bestAssemblyLock, currentBestAssembly, \
+                                    currentBestAssemblyScore, assembliesList) default(none)
+        for (unsigned assemblyID = 0; assemblyID < assembliesList.size(); assemblyID++) {
+            auto iterTuple = assembliesList.at(assemblyID);
```

Comment:
can you use pair destructuring here?

```
auto [current, score] = assembliesList.at(assemblyID)
```

Reply:
No, omp doesn't support this. A for loop with random access and comparison abort criteria is required, i.e. for(int i = 0; i < X; i++)

Reply:
No I don't mean using it in the for loop declaration but afterwards when assigning to `iterTuple`. OMP should not have anything to do with that right?

Reply:
ah gotcha. sure